pkgs/applications/office/roam-research/default.nix

```diff
@@ -0,0 +1,111 @@
+{ stdenv,
+  lib,
+  fetchurl,
+  alsa-lib,
+  atk,
+  cairo,
+  cups,
+  dbus,
+  dpkg,
+  expat,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  libX11,
+  libXScrnSaver,
+  libXcomposite,
+  libXcursor,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXi,
+  libXrandr,
+  libXrender,
+  libXtst,
+  libdrm,
+  libpulseaudio,
+  libxcb,
+  libxkbcommon,
+  libxshmfence,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  udev,
+}:
+
+let
+  libPath = lib.makeLibraryPath [
+    alsa-lib
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    gdk-pixbuf
+    glib
+    gtk3
+    libX11
+    libXcomposite
+    libXdamage
+    libXext
+    libXfixes
+    libXi
+    libXrandr
+    libdrm
+    libxcb
+    libxkbcommon
+    libxshmfence
+    mesa
+    nspr
+    nss
+    pango
+    stdenv.cc.cc
+    libXScrnSaver
+    libXcursor
+    libXrender
+    libXtst
+    libpulseaudio
+    udev
+ ];
+in
+stdenv.mkDerivation rec {
+  pname = "roam-research";
+  version = "0.0.15";
+
+  src = fetchurl {
+    url = "https://roam-electron-deploy.s3.us-east-2.amazonaws.com/${pname}_${version}_amd64.deb";
+    sha256 = "sha256-bcZuAsYaqbxTywxDuoALBzzr/0z6fEwttOfNlLD1a+w=";
+  };
+
+  nativeBuildInputs = [ dpkg ];
+
+  unpackPhase = ''
+    mkdir pkg
+    dpkg-deb -x $src pkg
+    sourceRoot=pkg
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    mv opt "$out/"
+
+    ln -s "$out/opt/Roam Research/roam-research" "$out/bin/roam-research"
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" --set-rpath "${libPath}:$out/opt/Roam Research:\$ORIGIN" "$out/opt/Roam Research/roam-research"
```

Comment:
You may use `autoPatchelfHook`

Reply:
@NickCao I tried using `autoPatchelfHook` however the binary coredumps, was not successful in identifying exactly why. These are the logs from console

```
APPIMAGE env is not defined, current application is not an AppImage
Starting server on port 3333
Listening on port 3333
[23450:0704/225542.944439:ERROR:object_proxy.cc(623)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[23450:0704/225542.944476:ERROR:select_file_dialog_linux_portal.cc(274)] Failed to read portal version property
[1]    23450 trace trap (core dumped)  ./result/bin/roam-research
[23453:0100/000000.452793:ERROR:zygote_linux.cc(622)] Zygote could not fork: proc
```